src/types.ts

```diff
@@ -25,3 +25,25 @@ export enum SelectionType {
   Hide,
   Highlight,
 }
+
+export enum AgentLevel {
+  Agent,
+  ChildAgent,
+}
+
+interface BaseCustomCheckboxProps {
+  agentName: string;
+  agentLevel: AgentLevel;
+}
+
+export interface ParentAgentProps extends BaseCustomCheckboxProps {
+  agentLevel: AgentLevel.Agent;
+  childName?: never;
```

Comment:
With a previous comment what I meant was why isn't this `childName?: string`

Reply:
I'm confused about this again: shouldn't a parent have the possibility of multiple children? so shouldn't this be `children?: string[]`?  or `childrenNames?: string[];